.github/workflows/soak-testing.yml

```diff
@@ -0,0 +1,302 @@
+# Pre-requisites:
+# - AWS Account with correct permissions
+# - `gh-pages` branch
+# - AWS Account needs to add the LOG_GROUP_NAME defined below
+
+name: Soak Testing
+on:
+  workflow_dispatch:
+    inputs:
+      target_commit_sha:
+        description: 'The commit SHA on this repo to use for the Soak Tests.'
+        required: true
+      test_duration_minutes:
+        description: 'The duration of the Soak Tests in minutes.'
+        required: true
+        default: 300
+  schedule:
+    - cron: '0 15 * * *'
+env:
+  # NOTE: The configuration of `APP_PROCESS_EXECUTABLE_NAME` is repo dependent
+  APP_PROCESS_EXECUTABLE_NAME: java
+  AWS_DEFAULT_REGION: us-east-1
+  AWS_WEB_IDENTITY_TOKEN_FILE: /tmp/awscreds
+  DEFAULT_TEST_DURATION_MINUTES: 300
+  HOSTMETRICS_INTERVAL_SECS: 600
+  CPU_LOAD_THRESHOLD: 71
+  TOTAL_MEMORY_THRESHOLD: 2147483648
+  MAX_BENCHMARKS_TO_KEEP: 100
+  # TODO: We might be able to adapt the "Soak Tests" to be "Overhead Tests".
+  # This means monitoring the Sample App's performance using high levels of TPS
+  # for the Load Generator over a shorter period of testing time. For example:
+  # https://github.com/aws-observability/aws-otel-collector/blob/main/docs/performance_model.md
+  # THROUGHPUT_PER_SECOND: TBD?
+
+jobs:
+  build_apps_to_ecr:
+    name: Publish Apps to ECR
+    runs-on: ubuntu-latest
+    steps:
+    # MARK: - GitHub Workflow Event Type Specific Values
+
+      - name: Use INPUT as commit SHA
+        if: ${{ github.event_name == 'workflow_dispatch' }}
+        run: |
+          echo "TARGET_SHA=${{ github.event.inputs.target_commit_sha }}" | tee --append $GITHUB_ENV;
+      - name: Use LATEST as commit SHA
+        if: ${{ github.event_name != 'workflow_dispatch' }}
+        run: |
+          echo "TARGET_SHA=${{ github.sha }}" | tee --append $GITHUB_ENV;
+
+    # MARK: - Publish Apps
+
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: ${{ env.TARGET_SHA }}
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 14
+      - uses: gradle/wrapper-validation-action@v1
+      - name: Configure AWS Credentials
+        run: |
+          export AWS_ROLE_ARN=${{ secrets.AWS_ASSUME_ROLE_ARN }}
+          echo AWS_ROLE_ARN=$AWS_ROLE_ARN >> $GITHUB_ENV
+          curl -H "Authorization: bearer $ACTIONS_ID_TOKEN_REQUEST_TOKEN" "$ACTIONS_ID_TOKEN_REQUEST_URL&audience=sigstore" | jq -r '.value' > $AWS_WEB_IDENTITY_TOKEN_FILE
+      - name: Login to ECR1
+        run: aws ecr-public get-login-password | docker login --username AWS --password-stdin public.ecr.aws
+      - name: Build snapshot with Gradle
+        uses: burrunan/gradle-cache-action@v1
+        with:
+          arguments: build snapshot --stacktrace -PenableCoverage=true
+        env:
+          PUBLISH_USERNAME: ${{ secrets.PUBLISH_USERNAME }}
+          PUBLISH_PASSWORD: ${{ secrets.PUBLISH_PASSWORD }}
```

Comment:
Yikes

Reply:
We discussed offline that the Soak Tests Design originally had a fundamental flaw. We should not be trying to publish snapshots during a Soak Test. We should just pull the image created during the Main build (or build a new image in the Soak Tests ourselves that is just like the main build).

The PR to fix this in the ADOT Python Downstream is here: https://github.com/aws-observability/aws-otel-python/pull/26

We will update this PR to match the result in that repo.